woocommerce/woocommerce-google-analytics-integration
# Implement JS Build (ES6) and JS Lint

### Changes proposed in this Pull Request:

Closes part of #227  .

This PR implements the next major changes.

- Add wp scripts and a basic webpack config for compiling purposes.
- source files are now in `assets/js/src` and the build in  `assets/js/build`
- Exclude source files `assets/js/src` to be included in archived zip file 
- Add js lint support as well as a github workflow for checking so on each PR
- Copy the Blocks actions present in https://github.com/woocommerce/woocommerce-blocks/pull/3953

### Checks:


### Screenshots:

### Detailed test instructions:


1. Run `npm i`
2. Run `npm run start`
3. Check the extension loads the `ga-integration.js` file `ga-admin-settings.js` file and they work correctly without errors.
4. Go to one of the files in `assets/js/src` and add a random console.log
5. See how the re-build process runs 
6. Refresh the page and see the console log running
7. Check the es-lint workflow is working fine
8. Try to make some bad changes like undefined constants or bad spacing and see the js linter working.
9. run `npm run archive`
10. See the zip file doesn't contain anymore `asset/js/src` but yes `asset/js/build`
11. Verify the config files 


### Additional details:

⚠️  JS Unit Tests support are not included in this PR
⚠️  A solution for dealing with duplicate actions with blocks will be done in the future PR
⚠️  To discuss if we want to add TypeScript on this project. Seems like a good moment and extension to do so.


### Changelog entry

> Dev - Implement JS Build (ES6) and JS Lint
